src/isomorphism/vf2.rs

```diff
@@ -765,76 +787,228 @@ where
+                                open_list: ol,
+                            };
+
+                            self.stack.push(f0);
+                            self.stack.push(Frame::Outer);
+                            continue;
+                        }
+                        Vf2Algorithm::<Ty, F, G>::pop_state(
+                            &mut self.st,
+                            nodes,
+                        );
+                    }
+                    match Vf2Algorithm::<Ty, F, G>::next_from_ix(
+                        &mut self.st,
+                        nodes[0],
+                        ol,
+                    ) {
+                        None => continue,
+                        Some(nx) => {
+                            let f = Frame::Inner {
+                                nodes: [nx, nodes[1]],
+                                open_list: ol,
+                            };
+                            self.stack.push(f);
+                        }
                     }
                 }
             }
         }
+        Ok(None)
     }
-    Ok(None)
 }
+
+macro_rules! vf2_mapping_impl {
+    ($name:ident, $Ty:ty) => {
+        #[pyclass(module = "retworkx", gc)]
+        pub struct $name {
+            vf2: Vf2Algorithm<$Ty, Option<PyObject>, Option<PyObject>>,
+        }
+
+        impl $name {
+            pub fn new(
+                py: Python,
+                g0: &StablePyGraph<$Ty>,
+                g1: &StablePyGraph<$Ty>,
+                node_match: Option<PyObject>,
+                edge_match: Option<PyObject>,
+                id_order: bool,
+                ordering: Ordering,
+                induced: bool,
+                call_limit: Option<usize>,
+            ) -> Self {
+                let vf2 = Vf2Algorithm::new(
+                    py, g0, g1, node_match, edge_match, id_order, ordering,
+                    induced, call_limit,
+                );
+                $name { vf2 }
+            }
+        }
+
+        #[pyproto]
+        impl PyIterProtocol for $name {
+            fn __iter__(slf: PyRef<Self>) -> Py<$name> {
+                slf.into()
+            }
+
+            fn __next__(
+                mut slf: PyRefMut<Self>,
+            ) -> PyResult<IterNextOutput<NodeMap, &'static str>> {
+                Python::with_gil(|py| match slf.vf2.next(py)? {
+                    Some(mapping) => Ok(IterNextOutput::Yield(mapping)),
+                    None => Ok(IterNextOutput::Return("Ended")),
+                })
+            }
+        }
+
+        #[pyproto]
+        impl PyGCProtocol for $name {
+            fn __traverse__(
+                &self,
+                visit: PyVisit,
+            ) -> Result<(), PyTraverseError> {
+                for j in 0..2 {
+                    for node in
+                        self.vf2.st[j].graph.node_indices().map(|node| {
+                            self.vf2.st[j].graph.node_weight(node).unwrap()
+                        })
```

Comment:
Would it be easier to just iterate over[ `node_weights()`](https://docs.rs/petgraph/0.6.0/petgraph/stable_graph/struct.StableGraph.html#method.node_weights) here instead of mapping the indices? Something like:

```suggestion
                        self.vf2.st[j].graph.node_weights()
```

Reply:
yes, it's definitely easier. i just copy+pasted this from `graph.rs`.